crates/client/src/lib.rs

```diff
@@ -310,109 +312,161 @@ impl<R: RegistryStorage, C: ContentStorage, N: NamespaceMapStorage> Client<R, C,
-Do you wish to initialize `{package_name}` and publish release y/N\n",
-package_name = &info.name,
-)
-                    } else {
-                        format!(
-"Package `{package_name}` does not already exist or you do not have access.
-You may be required to login. Try: `warg login`
-Do you wish to initialize `{package_name}` and publish release y/N\n",
-package_name = &info.name,
-)
-                    };
-                    if Confirm::with_theme(&ColorfulTheme::default())
-                        .with_prompt(prompt)
-                        .default(false)
-                        .interact()
-                        .unwrap()
-                    {
-                        info.entries.insert(0, PublishEntry::Init);
-                    } else {
-                        return Err(ClientError::MustInitializePackage {
-                            name,
-                            has_auth_token,
+        );
+        tracing::debug!("entries: {:?}", publish_info.entries);
+
+        let mut accepted_prompt_to_initialize = false;
+        let mut init_record_id: Option<RecordId> = None;
+
+        let (package, record) = loop {
+            let mut info = publish_info.clone();
+            let mut initializing = info.initializing();
+
+            let package = match self.fetch_package(&info.name).await {
+                Ok(package) => {
+                    if initializing {
+                        return Err(ClientError::CannotInitializePackage {
+                            name: package.name,
+                            init_record_id,
                         });
+                    } else if info.head.is_none() {
+                        // If we're not initializing the package and a head was not explicitly specified,
+                        // set to the latest known head.
+                        info.head = package.state.head().as_ref().map(|h| h.digest.clone());
                     }
+                    package
                 }
-                PackageInfo::new(info.name.clone())
-            }
-            err => err?,
-        };
-        let registry_domain = self.get_warg_registry(package.name.namespace()).await?;
+                Err(ClientError::PackageDoesNotExist {
+                    name,
+                    has_auth_token,
+                }) => {
+                    if !initializing {
+                        let prompt = if has_auth_token {
+                            format!(
+                                "Package `{package_name}` was not found.
+If it exists, you may not have access.
+Attempt to create `{package_name}` and publish the release y/N\n",
+                                package_name = &info.name,
+                            )
+                        } else {
+                            format!(
+                                "Package `{package_name}` was not found.
+If it exists, you may not have access without logging in. Try: `warg login`
```

Comment:
Should we suggest to users that they log in and immediately prompt them to publish without making them log in?  Also guessing that if the registry they're using requires auth, that that'll show up in the error?  Not sure what best DX is here, but might want to consider just using the same prompt for both cases